modules/home/workstation/i3/default.nix

```diff
@@ -2,19 +2,65 @@
 
 with lib;
 
-{
-  options.mine.workstation.i3.enable = mkEnableOption "workstation.i3";
+let
+  cfg = config.mine.workstation.i3;
+in {
 
-  config = mkIf config.mine.workstation.i3.enable {
+  options.mine.workstation.i3 = {
+    enable = mkEnableOption "workstation.i3";
+
+    bar = {
+      engine = mkOption {
+        type = types.enum [ "i3bar" "polybar" ];
+        default = "i3bar";
+        description = ''
+          Select the bar to use with i3
+        '';
+      };
+
+      battery = {
+        device = mkOption {
+          default = "BAT0";
+          description = ''
+            Battery to be monitored by the bar engine.
+          '';
+        };
+        fullAt = mkOption {
+          default = 98;
+          description = ''
+            In case the battery never reports 100% charge.
+          '';
+        };
+      };
+
+      wlan = mkOption {
+        type = types.str;
+        default = "";
+        description = ''
+          WLAN interface to be monitored by the bar engine.
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
     assertions = [
       {
         assertion = config.mine.nixosConfig != {} && config.mine.darwinConfig == {};
         message = "mine.workstation.i3.enable must be false on Darwin!";
       }
+      {
+        assertion = (cfg.bar.engine == "polybar" && config.xsession.windowManager.i3.config.bars == []) || (cfg.bar.engine == "i3bar" && ((builtins.length config.xsession.windowManager.i3.config.bars) != 0));
+        message = "There must be no i3bars if polybar is enabled.";
```

Comment:
The error message here is going to be confusing if `cfg.bar.engine = "i3bar"` and `config.xsession.windowManager.i3.config.bars = []`. You probably should split this assertion into two separate ones.

Reply:
How would you split them ?

Reply:
Actually, let's leave this like this. This will be solved in an upcoming PR